ir/src/main/scala/org/scalajs/core/ir/Infos.scala

```diff
@@ -356,6 +359,9 @@ object Infos {
         case ClassOf(cls) =>
           builder.addAccessedClassData(cls)
 
+        case JSLoadConstructor(cls) =>
+          builder.addInstantiatedClass(cls.className)
+
```

Comment:
This is actually funny :P
In the case where we use a class only in an `isInstanceOf`, if we don't use this over approximation, the unoptimized code will fail, but optimized code will be able to elide the check entirely...

I guess leaving it for now is fine then :)